components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/handlers/util/ResponseTypeHandlerUtil.java

```diff
@@ -253,9 +254,15 @@ public static AuthzCodeDO generateAuthorizationCode(OAuthAuthzReqMessageContext
         // set code issued time.this is needed by downstream handlers.
         oauthAuthzMsgCtx.setCodeIssuedTime(timestamp.getTime());
 
-        if (authorizationReqDTO.getUser() != null && authorizationReqDTO.getUser().isFederatedUser()) {
-            //if a federated user, treat the tenant domain as similar to application domain.
-            authorizationReqDTO.getUser().setTenantDomain(authorizationReqDTO.getTenantDomain());
+        AuthenticatedUser authenticatedUser = authorizationReqDTO.getUser();
+        if (authenticatedUser != null && authenticatedUser.isFederatedUser()) {
+            String appName = OAuth2Util.getServiceProvider(authorizationReqDTO.getConsumerKey()).getApplicationName();
```

Comment:
Getting the whole SP object to just get the app name is not optimal (results in multiple db queries).

Can we introduce a method that only fetches the application name only (if not already available)?

Reply:
or Introduce a method that only fetches the basic application info and gets the app name from it?

Reply:
Ack, introduce new util methods to fetch only application name using application management service.